### ISSUE TITLE
SA-FARI SAM2 fine-tune for animal segmentation

### ISSUE THREAD

**Issue by babajideowoyele** [MaskAnyone/MaskAnyone#75]
## Summary

Fine-tune SAM2 on Meta's SA-FARI animal video dataset to improve segmentation quality on primate/animal footage, particularly field recordings with occlusion, foliage, and variable lighting.

## Motivation

SAM2 out-of-the-box is trained primarily on SA-1B (human and object data). Segmentation quality degrades on animal subjects in natural-habitat conditions. A fine-tuned checkpoint is expected to substantially improve mask IoU on such footage.

## References

- SA-FARI dataset: https://huggingface.co/datasets/facebook/SA-FARI
- SAM2 training code: `sam2/training/` inside the container at `/workspace/segment-anything-2/training/`

## Tasks

- [ ] Download and inspect SA-FARI dataset structure
- [ ] Set up SAM2 training pipeline inside the Docker container
- [ ] Run fine-tuning on SA-FARI (start with `sam2.1_hiera_tiny` for iteration speed)
- [ ] Add fine-tuned checkpoint to `MODEL_CONFIGS` in `sam2/src/segmentation.py`
- [ ] Expose fine-tuned model in frontend segmentation model dropdown
- [ ] Benchmark: baseline SAM2 vs SA-FARI fine-tune on representative primate footage — report mask IoU

## Integration point

```python
# sam2/src/segmentation.py
"sam2.1_hiera_tiny_safari": {
    "checkpoint": "/workspace/sam2/checkpoints/sam2.1_hiera_tiny_safari.pt",
    "config": "configs/sam2.1/sam2.1_hiera_t.yaml",
},
```

## Branch

`feat/primate-workflow`
